### PR TITLE
Add deeparg (no docs!)

### LINKED ISSUE #5: Add deepARG
### Description of feature

Add deepARG module

[transferred from old repository]

### PR DESCRIPTION
**Pull request by jfy133**
This adds DeepARG to close #5 

Does not include docs as we will do this later.

Code doesn't break, but hasn't been fully tested because no compatible test data yet.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
    - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/funcscan/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/funcscan _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: bin/check_samplesheet.py [hunk 1]
```diff
@@ -92,9 +92,9 @@ def check_samplesheet(file_in, file_out):
                 if fasta:
                     if fasta.find(" ") != -1:
                         print_error("FastA file contains spaces!", "Line", line)
-                    if not fasta.endswith(".fasta.gz") and not fasta.endswith(".fa.gz") and not fasta.endswith(".fna.gz") and not fasta.endswith(".fna"):
+                    if not fasta.endswith(".fasta.gz") and not fasta.endswith(".fa.gz") and not fasta.endswith(".fna.gz")  and not fasta.endswith(".fasta") and not fasta.endswith(".fa") and not fasta.endswith(".fna"):
```
Comment (Midnighter) [inline bin/check_samplesheet.py:95]: Could change this to:

```suggestion
                    if Path(fasta).suffixes.join("") not in (".fasta.gz", ".fa.gz", ".fna.gz", ".fasta", ".fa", ".fna"):
```

File: nextflow.config [hunk 2]
```diff
@@ -21,6 +21,16 @@ params {
     // fARGene options
     fargene_hmm_model          = 'class_a'
 
+    // DeepARG options
+    deeparg_data                        = null
+    deeparg_model                       = 'ss'
```
Comment (jasmezz) [inline nextflow.config:26]:
```suggestion
    deeparg_model                       = 'SS'
```
See documentation of deepARG

**Review by jasmezz — APPROVED** [review]

File: bin/check_samplesheet.py [hunk 3]
```diff
@@ -92,9 +92,9 @@ def check_samplesheet(file_in, file_out):
                 if fasta:
                     if fasta.find(" ") != -1:
                         print_error("FastA file contains spaces!", "Line", line)
-                    if not fasta.endswith(".fasta.gz") and not fasta.endswith(".fa.gz") and not fasta.endswith(".fna.gz") and not fasta.endswith(".fna"):
+                    if Path(fasta).suffixes.join("") not in (".fasta.gz", ".fa.gz", ".fna.gz", ".fasta", ".fa", ".fna"):
```
Comment (jfy133) [inline bin/check_samplesheet.py:95]:
```suggestion
                    if not fasta.endswith(".fasta.gz") and not fasta.endswith(".fa.gz") and not fasta.endswith(".fna.gz")  and not fasta.endswith(".fasta") and not fasta.endswith(".fa") and not fasta.endswith(".fna"):
```

File: bin/check_samplesheet.py (hunk 1 above)
Comment (jfy133) [reply bin/check_samplesheet.py:95]: Was just following standard template, could do but make check with Harshil that there wasn't a specific reason why he didn't the  library that Path is from already

**Pull request merged**